app/src/main/java/com/olayg/shibeonline/view/MainActivity.kt

```diff
@@ -1,26 +1,83 @@
 package com.olayg.shibeonline.view
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.util.Log
 import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.olayg.shibeonline.R
+import com.olayg.shibeonline.adapter.ShibeAdapter
+import com.olayg.shibeonline.databinding.ActivityMainBinding
 import com.olayg.shibeonline.viewmodel.ShibeViewModel
 
 class MainActivity : AppCompatActivity() {
 
+    /**
+     * ViewModel for shibe API
+     */
     private val viewModel by viewModels<ShibeViewModel>()
 
+    /**
+     * Layout binding
+     */
+    private val binding by lazy { ActivityMainBinding.inflate(layoutInflater) }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        setContentView(binding.root)
 
-        // Use this method to get the images
-        viewModel.getImages(10)
+        initViews()
 
+        // Observe shibes data
         viewModel.shibes.observe(this) {
-            // Here is where your will get the result
-            Log.d("MainActivity", "onCreate: $it")
+
+            // Update shibes with new data
+            (binding.rvImages.adapter as ShibeAdapter).updateUrls(it)
+
+        }
+
+    }
+
+    /**
+     * Initialize views and listeners
+     */
+    private fun initViews() = with(binding) {
+
+        // Listener for count input
+        btnFetch.setOnClickListener {
+
+            val count = binding.etCount.text?.toString()?.toIntOrNull()
+
+            viewModel.getImages(count ?: 1)
+
         }
+
+        btnSwitchLayout.setOnClickListener {
+
+            // If layoutManager is an instance of LinearLayoutManager
+            if (rvImages.layoutManager!!::class.java == LinearLayoutManager::class.java) {
```

Comment:
We shouldn't use this `!!` to force the `layoutManager` as a non null property

Reply:
I didn't like how I did that as well.
I ended up casting the layoutManager to LayoutManager but I'm not sure if I'm satisfied with that or not.

Reply:
I switched to `rvImages.layoutManager is GridLayoutManager` since `is LinearLayoutManager` won't work because GridLayoutManager extends LinearLayoutManager.